mailpoet/lib/Automation/Integrations/MailPoet/Triggers/OrderStatusChangedTrigger.php

```diff
@@ -0,0 +1,152 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Automation\Integrations\MailPoet\Triggers;
+
+use MailPoet\Automation\Engine\Data\StepRunArgs;
+use MailPoet\Automation\Engine\Data\StepValidationArgs;
+use MailPoet\Automation\Engine\Data\Subject;
+use MailPoet\Automation\Engine\Hooks;
+use MailPoet\Automation\Engine\Integration\Trigger;
+use MailPoet\Automation\Integrations\MailPoet\Payloads\OrderStatusChangePayload;
+use MailPoet\Automation\Integrations\MailPoet\Subjects\CustomerSubject;
+use MailPoet\Automation\Integrations\MailPoet\Subjects\OrderStatusChangeSubject;
+use MailPoet\Automation\Integrations\MailPoet\Subjects\OrderSubject;
+use MailPoet\Automation\Integrations\MailPoet\Subjects\SubscriberSubject;
+use MailPoet\Entities\SubscriberEntity;
+use MailPoet\Segments;
+use MailPoet\Subscribers\SubscribersRepository;
+use MailPoet\Validator\Builder;
+use MailPoet\Validator\Schema\ObjectSchema;
+use MailPoet\WooCommerce;
+use MailPoet\WP\Functions;
+
+class OrderStatusChangedTrigger implements Trigger {
+
+
+  /** @var Functions */
+  private $wp;
+
+  /** @var SubscribersRepository */
+  private $subscribersRepository;
+
+  /** @var Segments\WooCommerce */
+  private $woocommerce;
+
+  /** @var WooCommerce\Helper */
+  private $woocommerceHelper;
+
+  public function __construct(
+    Functions $wp,
+    SubscribersRepository $subscribersRepository,
+    Segments\WooCommerce $wooCommerce,
+    WooCommerce\Helper $woocommerceHelper
+  ) {
+    $this->wp = $wp;
+    $this->subscribersRepository = $subscribersRepository;
+    $this->woocommerce = $wooCommerce;
+    $this->woocommerceHelper = $woocommerceHelper;
+  }
+
+  public function getKey(): string {
+    return 'mailpoet:order-status-changed';
+  }
+
+  public function getName(): string {
+    return __('Order status changed', 'mailpoet');
+  }
+
+  public function getArgsSchema(): ObjectSchema {
+
+    return Builder::object([
+      'from' => Builder::string()->required()->default('any'),
+      'to' => Builder::string()->required()->default('wc-completed'),
+    ]);
+  }
+
+  public function getSubjectKeys(): array {
+    return [
+      SubscriberSubject::KEY,
+      OrderSubject::KEY,
+      OrderStatusChangeSubject::KEY,
+      CustomerSubject::KEY,
+    ];
+  }
+
+  public function validate(StepValidationArgs $args): void {
+    // TODO: Implement validate() method.
```

Comment:
I guess we should remove `TODO` if it needs no further validation.

Reply:
gone.